drivers/adc/adc_stm32.c

```diff
@@ -369,14 +383,57 @@ static void adc_stm32_calib(const struct device *dev)
 	};
 #endif /* ! ADC_VER_V5_V90 */
 
+	/*
+	 * Function to configure the oversampling scope. It is basically a
+	 * wrapper over LL_ADC_SetOverSamplingScope() which in addition stops
+	 * the ADC if needed.
+	 */
+static void adc_stm32_oversampling_scope(ADC_TypeDef *adc, uint32_t ovs_scope)
+{
+#if defined(CONFIG_SOC_SERIES_STM32L0X) || \
+	defined(CONFIG_SOC_SERIES_STM32WLX)
+	/*
+	 * setting OVS bits is conditioned to ADC state: ADC must be disabled
+	 * or enabled without conversion on going : disable it, it will stop
+	 */
+	if (LL_ADC_GetOverSamplingScope(adc) == ovs_scope) {
+		return;
+	}
+	adc_stm32_disable(adc);
+#endif
+	LL_ADC_SetOverSamplingScope(adc, ovs_scope);
+}
+
+	/*
+	 * Function to configure the oversampling scope. It is basically a
+	 * wrapper over LL_ADC_SetOverSamplingRatioShift() which in addition
+	 * stops the ADC if needed.
+	 */
```

Comment:
Ditto

Reply:
Thanks for the review. I have basically indented it that way to match what is done below for `adc_stm32_oversampling()`, presumably because it is inside a `#ifdef` `#endif` block.

Anyway I have just fixed both issues and pushed that.